woocommerce/woocommerce
# Improve the prompt of product title suggestions

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR has some minor improvements to the prompt and JSON retrieval process. I’ve removed whitespace and shortened the prompt to make it faster. With these changes, LLM returns responses in less than 8s (usually 7s), down from 15s.

Moreover, I tried getting a CSV response from the model, but it didn’t improve the response time. So, I switched back to JSON.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use the product title suggestion feature to generate suggestions (if you don't have the project set up, follow the instructions [here](https://github.com/woocommerce/woocommerce/pull/38452))
2. Confirm that suggestions are generated and the response time is faster.
